src/SBH_LinearRegressionChart.js

```diff
@@ -0,0 +1,150 @@
+// src/SBH_LinearRegressionChart.js
+import React from "react";
+import styled from "styled-components";
+
+import ReactResizeDetector from "react-resize-detector";
+
+import Highcharts from "highcharts";
+
+import UUIDv1 from "uuid/v1";
+
+const Root = styled.div`
+  width: 100%;
+  height: 100%;
+`;
+
+class SBH_LinearRegressionChart extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.chart = {};
+
+    this.state = {
+      chartId: UUIDv1(),
+    };
+  }
+
+  componentDidMount() {
+    this.chart = Highcharts.chart(this.state.chartId, {
+      title: {
+        text: this.props.title,
+      },
+
+      xAxis: {
+        title: {
+          text: this.props.xAxisDataKey,
+        },
+      },
+
+      yAxis: {
+        title: {
+          text: this.props.yAxisDataKey,
+        },
+      },
+
+      plotOptions: {
+        series: {
+          label: {
+            connectorAllowed: false,
+          },
+        },
+      },
+
+      series: [
+        {
+          type: "line",
+          name: this.props.yAxisDataKey,
+          data: this.jsonArrayToLineData(this.props.data),
+        },
+        {
+          type: "scatter",
+          name: this.props.xAxisDataKey,
+          data: this.jsonArrayToScatterData(this.props.data),
+          marker: {
+            radius: 4,
+          },
+        },
+      ],
+
+      responsive: {
+        rules: [
+          {
+            condition: {
+              maxWidth: 500,
+            },
+            chartOptions: {
+              legend: {
+                align: "center",
+                verticalAlign: "bottom",
+                layout: "horizontal",
+              },
+              yAxis: {
+                labels: {
+                  align: "left",
+                  x: 0,
+                  y: -5,
+                },
+                title: {
+                  text: null,
+                },
+              },
+              subtitle: {
+                text: null,
+              },
+              credits: {
+                enabled: false,
+              },
+            },
+          },
+        ],
+      },
+
+      credits: {
+        enabled: false,
+      },
+    });
+  }
+
+  onResize = (width, height) => {
+    this.chart.setSize(width, height, false);
+  };
+
+  jsonArrayToLineData = (jsonArray) => {
+    var regressionlineData = [];
+    jsonArray[0].regressionlineData.map((jsonObject) => {
```

Comment:
Why are you using only the first element of this.props.data(jsonArray[0])?
I think it will be a good to divide regressionLineData and scatterData.

Reply:
Those data were divided